docs/rag/adrs/paragon-dependency.md

```diff
@@ -0,0 +1,24 @@
+
+Note that the name PaRAGon is preliminary and may change to avoid conflicts with third-party trademarks.  This document will continue to refer to the asset as PaRAGon below, but when it does so it really means "the asset that is currently called PaRAGon but might be called something different in the future".
+
+### Alternatives
+
+- InstructLab could follow the model in the draft pull request indefinitely.  That would give InstructLab more flexibility to focus on InstructLab-specific needs.  However, it would also lead to a lot of duplicated effort with the team working on PaRAGon, which already pays a lot of attention to InstructLab-specific needs.
+- InstructLab could adopt an off-the-shelf RAG platform instead of PaRAGon.  For example, PaRAGon itself is built on Haystack; we could just depend on Haystack directly.  However, Haystack is an very flexible platform for developers, so adopting Haystack directly in InstructLab would entail making a lot of technical decisions that have considerable impact on the effectiveness of the solution.  So either we would make the same decisions that PaRAGon makes (which brings us back to the option above) or we would make different decisions (which would mean even more duplicated effort plus substantial inconsistency).
+- InstructLab could fork its own copy of PaRAGon and put that in the InstructLab organization in github (either in an existing repository there or a new one).  That would make it a lot easier to incorporate changes from PaRAGon than just having code inspired by PaRAGon, because we could just copy the entire code-base over when we want to update the fork.  However, it still seems a lot clunkier than just having a dependency on PaRAGon.
+
+## Decision
```

Comment:
The general guidance is to try each of these documents be about a single decision. I think the one being documented here is that

```
InstructLab will directly depend on Paragon in the 1.5 release`.
```

The other two line items here become consequences, something like:

```
* The initial RAG implementation for 1.4 will be a temporary, bespoke implementation that is only inspired by PaRAGon.
* We will have to remove that temporary implementation after the release of 1.4.
```